EKF/control.cpp

```diff
@@ -876,6 +878,19 @@ void Ekf::controlHeightSensorTimeouts()
 				failing_height_source = "ev";
 				new_height_source = "ev";
 
+			// Fallback to rangefinder data if available
+			} else if (_range_sensor.isHealthy()) {
+				setControlRangeHeight();
+				if (_control_status.flags.in_air && isTerrainEstimateValid()) {
+					_hgt_sensor_offset = _terrain_vpos;
+				} else if (_control_status.flags.in_air) {
+				    _hgt_sensor_offset = _range_sensor.getDistBottom() + _state.pos(2);
+			    } else {
+					_hgt_sensor_offset = _params.rng_gnd_clearance;
+				}
```

Comment:
This part of the code shouldn't be there. I'm not sure if you can add it cleanly in `resetHeight` or somewhere else, but this function should only make the reset decision, nothing more complicated.

Reply:
I absolutely agree.

Reply:
_hgt_sensor_offset calculation moved to resetHeight with [4602349](https://github.com/PX4/PX4-ECL/pull/994/commits/4602349b289af0c448299dd4a14b3051a68c73f6).